app/controllers/user_controller.py

```diff
@@ -100,70 +115,77 @@ def update_user(id):
                 user.role = role
             if grade is not None:
                 user.grade = grade
+            if dob is not None:
+                user.dob = dob
             user.updated_at = updated_at
 
             db.session.commit()
 
-            return 'Data user has been updated'
-
+            return response.success("", "data user has been updated")
         else:
-            return 'Update data failed'
-    
+            return "Update data user is failed"
```

Comment:
this is grammatically incorrect.
Should be:
`Failed updating user data`

I think it would be better:
`Failed updating user data, user not found.`

And don't forget to set the status code as well, make sure the return is consistent in each case.

Reply:
Could be like the one in delete_user:
` return response.badRequest([], "User not found")`